opm/simulators/wells/WellState.hpp

```diff
@@ -354,13 +358,13 @@ namespace Opm
         WellState& operator=(const WellState& rhs) = default;
 
     private:
+        PhaseUsage phase_usage_;
```

Comment:
Hm, this is a bit dangerous.  I think you'll have to implement user-defined special member functions (i.e., assignment, copy constructor and their "move" counterparts) now.  The [`PhaseUsage`]() structure does not have proper value semantics since it contains two [built-in arrays](https://github.com/OPM/opm-simulators/blob/708645a2c07bb232609bff80f3967d1dc625a060/opm/core/props/BlackoilPhases.hpp#L46-L47) instead of proper objects.

Reply:
OK - I'll use `std::array<>` in `PhaseUsage`